cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -241,19 +248,60 @@ compute_submap_indices(const IndexMap& imap,
     std::ranges::sort(global_idx_to_possible_owner);
 
     // Choose the submap owner for each index in `recv_indices`
+    // and pack destination ranks for each process that has received new indices
     std::vector<int> send_owners;
     send_owners.reserve(recv_indices.size());
-    for (auto idx : recv_indices)
+    std::vector<int> new_owner_dest_ranks;
+    new_owner_dest_ranks.reserve(recv_indices.size());
```

Comment:
Again, not sure if this reserve makes sense

Reply:
You need a minimal reserve for mpi communication.
Are you talking about the new owner dest ranks reservation?
We could reduce this to `.reserve(1)`, which would be the minimal case, where each re-assigned index is only required by the new owning process (no real determination of ownership).